apps/rewards/app/components/Content/MyRewards.js

```diff
@@ -1,27 +1,203 @@
 import PropTypes from 'prop-types'
 import React from 'react'
 import styled from 'styled-components'
-
+import {
+  Table,
+  TableHeader,
+  TableRow,
+  TableCell,
+  Text,
+  Button,
+  IdentityBadge,
+  Viewport,
+  theme,
+  IconCheck,
+  IconCross,
+  IconFundraising,
+  IconTime,
+  ContextMenu,
+  ContextMenuItem,
+} from '@aragon/ui'
+import { displayCurrency } from '../../utils/helpers'
+import {
+  AverageRewards,
+  formatAvgAmount,
+  RewardDescription,
+  RewardsTable,
+  NarrowList,
+  NarrowListReward,
+  AmountBadge,
+} from './RewardsTables'
 import { Empty } from '../Card'
+import { provideNetwork } from '../../../../../shared/ui'
+
+const averageRewardsTitles = [ 'My Unclaimed Rewards', 'Year to Date', 'Inception to Date' ]
+// TODO: these need to be actually calculated
+const averageRewardsNumbers = [
+  formatAvgAmount(146, '+$', 'green'),
+  formatAvgAmount(19989.88, '$'),
+  formatAvgAmount(799.87, '$'),
+]
+
+const generateOpenDetails = (reward, openDetails) => () => {
+  console.log('calling openDetails on', reward)
+  openDetails(reward)
+}
+
+const translateToken = (token) => {
+  if (token == 0x0) {
+    return 'ETH'
+  }
+}
+
+const MyRewardsWide = ({ claimed, data, openDetails, network }) => (
+  <Table
+    style={{ width: '100%' }}
+    header={
+      <TableRow>
+        <TableHeader key="1" title="Description" />
+        <TableHeader key="2" title="Transaction" />
+        <TableHeader key="3" title={claimed ? 'Status': 'Transaction Date'} />
+        <TableHeader key="4" title="Amount" />
+      </TableRow>
+    }
+  >
+    {data.map((reward, i) => (
+      <ClickableTableRow key={i} onClick={generateOpenDetails(reward, openDetails)}>
```

Comment:
Tried to follow the flow of props (specifically `openDetails`), but figured it be easier just to ask - are you meaning to invoke `generateOpenDetails` _inside_ the onClick? If so, `generateOpenDetails` should eventually return another function, not static data (so the `onClick` handler can invoke the function passed to it). 

Second pair of eyes in case you meant to do: `onClick={() => generateOpenDetails(reward, openDetails)}`

Reply:
Looks like this is the function:

https://github.com/AutarkLabs/planning-suite/blob/107a67f1c23eb37eb45268e1a0ceaf6c1442223d/apps/rewards/app/components/App/App.js#L184 

Is this working how you would expect it?

Reply:
> generateOpenDetails should eventually return another function

It does:

const generateOpenDetails = (reward, openDetails) => () => {